### ISSUE TITLE
Very low validation acuuracy with ImageDataGanarator and flow_from_directory

### ISSUE THREAD

**Issue by Emnaa12** [keras-team/keras#12912]
I am trying to train a 3D resnet-50 with stereo data, i.e. the input of the network is two RGB images (2,224,224,3) concatenated according to the depth axe to get 6 grayscale images (6,224,224,1). Since I am training my network on a large database, I used ImageDataGenerator flow_from_directory method to get labels from folders. My data are 22390 pairs divided into 2239 classes (10 images per class).
I have split them manually into 0.8 for training and 0.2 for validation through taking the two middle images from each class to the validation set. I did not use shuffling during the training because I have to take the right corresponding inputs of the stereo pairs from the two directories. My problem is that during the training, the accuracy of the training was increasing and the loss was decreasing which is good. In fact, the validation accuracy was very low (around 0.001) and there is no improvement. Also the valisation loss is very high and still oscillating into very high values!!
I wonder if that is due to:
- My data which contains a large number of classes with few images per class?
- The fact that I am not shuffling the data before or during the split?
- Other reason.

Here is my code:
   
     
      import numpy as np 
      from keras_preprocessing.image import ImageDataGenerator
      from resnet3d import Resnet3DBuilder
      import keras 

      data_path_l='.\TRAIN\left\'
      data_path_r='.\TRAIN\right\'
      test_data_path_l ='.\TEST\left\'
      test_data_path_r='.\TEST\right\'
     num_classes=2239
     batch_size=32
     epochs=500
     train_images=17912
    val_images=4478

    input_imgen = ImageDataGenerator()
    def generate_generator_multiple(generator,dir1, dir2, batch_size):
           genX1 = generator.flow_from_directory(directory=dir1,target_size=(224,224)
                                         color_mode="rgb",
                                          batch_size=batch_size,
                                          class_mode="categorical",
                                          shuffle=False)
                                          
     

         genX2 = generator.flow_from_directory(directory=dir2,target_size=(224,224),
                                          color_mode="rgb",
                                          batch_size=batch_size,
                                          class_mode="categorical",
                                          shuffle=False)
    
          while True:
                 X1i = genX1.next()
                 X2i = genX2.next()
                 Xsum = np.concatenate((X1i[0],X2i[0]), axis=3)
                 Xsum = np.expand_dims(Xsum, axis=1)
                 Xsum=np.swapaxes(Xsum,4,1)
                 yield Xsum, X2i[1]  #Yield both images and their mutual label
            
            
    def validation_generate_generator_multiple(generator,dir1, dir2, batch_size):
          val_genX1 = generator.flow_from_directory(directory=dir1,target_size=(224,224),
                                          color_mode="rgb",
                                          batch_size=batch_size,
                                          class_mode="categorical",
                                          shuffle=False)
                                          
      
          val_genX2 = generator.flow_from_directory(directory=dir2,target_size=(224,224),
                                          color_mode="rgb",
                                          batch_size=batch_size,
                                          class_mode="categorical",
                                          shuffle=False)
    
          while True:
                 X1i = val_genX1.next()
                 X2i = val_genX2.next()
                 Xsum = np.concatenate((X1i[0],X2i[0]), axis=3)
                 Xsum = np.expand_dims(Xsum, axis=1)
                Xsum=np.swapaxes(Xsum,4,1)
                yield Xsum, X2i[1]  #Yield both images and their mutual label
            
            
    inputgenerator=generate_generator_multiple(generator=input_imgen,
                                           dir1=data_path_l,
                                           dir2=data_path_r,
                                           batch_size=batch_size) 
  
 
 
    validation_inputgenerator= validation_generate_generator_multiple(generator=input_imgen,
                                           dir1=test_data_path_l,
                                           dir2=test_data_path_r,
                                           batch_size=batch_size)
                                                  
     
    model=Resnet3DBuilder.build_resnet_50((6,224,224,1), num_classes)
    model.compile(loss='categorical_crossentropy',
                            optimizer='adam',
                            metrics=['accuracy'])
     history=model.fit_generator(inputgenerator,
                        steps_per_epoch= train_images // batch_size,
                        validation_data = validation_inputgenerator, 
                        validation_steps = val_images// batch_size,
                        epochs = epochs,
                        shuffle=False)
    model.save('resnet3D_6sample_lr.h5')`

and here is a part from the training process

    
    Epoch 153/500
    559/559 [==============================] - 1927s 3s/step -                 
     loss: 1.1738 - acc: 0.9620 - val_loss: 12.9716 - val_acc: 0.0038

    Epoch 154/500
    559/559 [==============================] - 1929s 3s/step - 
    loss: 1.1810 - acc: 0.9594 - val_loss: 12.4732 - val_acc: 0.0065

    Epoch 155/500
    559/559 [==============================] - 2161s 4s/step - 
    loss: 1.1146 - acc: 0.9681 - val_loss: 11.9151 - val_acc: 0.0076

    Epoch 156/500
    559/559 [==============================] - 2165s 4s/step - 
    loss: 1.0637 - acc: 0.9734 - val_loss: 10.2526 - val_acc: 0.0135

    Epoch 157/500
    559/559 [==============================] - 2192s 4s/step - 
    loss: 1.0324 - acc: 0.9745 - val_loss: 9.6518 - val_acc: 0.0148

    Epoch 158/500
    559/559 [==============================] - 1921s 3s/step - 
    loss: 1.1294 - acc: 0.9516 - val_loss: 11.7690 - val_acc: 0.0090

    Epoch 159/500
    559/559 [==============================] - 1504s 3s/step - 
    loss: 1.1739 - acc: 0.9520 - val_loss: 11.2503 - val_acc: 0.0090

    Epoch 160/500
    559/559 [==============================] - 2139s 4s/step - 
    loss: 1.0292 - acc: 0.9732 - val_loss: 11.8762 - val_acc: 0.0106

    Epoch 161/500
    559/559 [==============================] - 1835s 3s/step - 
    loss: 1.0494 - acc: 0.9671 - val_loss: 10.1109 - val_acc: 0.0151

    Epoch 162/500
    559/559 [==============================] - 1583s 3s/step - 
    loss: 0.9970 - acc: 0.9737 - val_loss: 12.5503 - val_acc: 0.0049

**Comment by Alpkant** [comment]
I have the same issue, I train VGG16 with 90% accuracy but on the same data ResNet always give 0.001 validation accuracy. I gave the training input as both training and validation and accuracy is still 0.001.
